controllers/k8ssandra/cleanup.go

```diff
@@ -66,51 +60,6 @@ func (r *K8ssandraClusterReconciler) checkDeletion(ctx context.Context, kc *api.
 			logger.Error(err, "Failed to delete CassandraDatacenter", "CassandraDatacenter", dcKey, "Context", dcTemplate.K8sContext)
 			hasErrors = true
 		}
-
-		selector := k8ssandralabels.CleanedUpByLabels(kcKey)
-		stargateList := &stargateapi.StargateList{}
-		options := client.ListOptions{
-			Namespace:     namespace,
-			LabelSelector: labels.SelectorFromSet(selector),
-		}
-
-		err = remoteClient.List(ctx, stargateList, &options)
-		if err != nil {
-			logger.Error(err, "Failed to list Stargate objects", "Context", dcTemplate.K8sContext)
-			hasErrors = true
-			continue
-		}
-
-		for _, sg := range stargateList.Items {
-			if err = remoteClient.Delete(ctx, &sg); err != nil {
-				key := client.ObjectKey{Namespace: namespace, Name: sg.Name}
-				if !errors.IsNotFound(err) {
-					logger.Error(err, "Failed to delete Stargate", "Stargate", key,
-						"Context", dcTemplate.K8sContext)
-					hasErrors = true
-				}
-			}
-		}
-
-		if r.deleteReapers(ctx, kc, dcTemplate, namespace, remoteClient, logger) {
-			hasErrors = true
-		}
-
-		if r.deleteDeployments(ctx, kc, dcTemplate, namespace, remoteClient, logger) {
-			hasErrors = true
-		}
-
-		if r.deleteServices(ctx, kc, dcTemplate, namespace, remoteClient, logger) {
-			hasErrors = true
-		}
```

Comment:
The deployments and services cleanup was added for #980, but we changed the way Medusa is deployed since then, so this was already stale code.